core/src/fees.ts

```diff
@@ -23,16 +26,42 @@ export const getApproximateTransactionFees = async function (network: string): P
     };
 };
 
-export const enrichAuctionWithTransactionFees = function (
+export const enrichAuctionWithTransactionFees = async function (
     auction: Auction,
-    fees: TransactionFees
-): AuctionTransaction {
+    fees: TransactionFees,
+    network: string
+): Promise<AuctionTransaction> {
+    let totalFeeETH = new BigNumber(0);
+    let totalFeeDAI = new BigNumber(0);
+
+    try {
+        const signer = await getSigner(network);
+        const address = await signer.getAddress();
+        const isWalletAuthed = await getWalletAuthorizationStatus(network, address);
+        const isCollateralAuthed = await getCollateralAuthorizationStatus(network, auction.collateralType, address);
+
+        if (!isWalletAuthed) {
+            totalFeeETH = totalFeeETH.plus(fees.authTransactionFeeETH);
+            totalFeeDAI = totalFeeDAI.plus(fees.authTransactionFeeDAI);
```

Comment:
I suggest to use one currency and convert it afterwards

Reply:
I wanted to avoid this. In order to convert ETH to DAI I would need to refetch the exchange rate.

```js
const convertETHtoDAI = function (eth: BigNumber) {
    return eth.multipliedBy(exchangeRate);
};
```

However an alternative idea I had was to put the exchangeRate into the fees object. Then I could use it somewhere else. What do you think?

Reply:
> I would need to refetch the exchange rate

This one suppose to be  cached if you don't provide an `amount` argument